common-lib/src/main/scala/com/gu/mediaservice/lib/imaging/ImageOperations.scala

```diff
@@ -114,31 +116,60 @@ class ImageOperations(playPath: String) extends GridLogging {
+    val cropSource  = addImage(sourceFile)
+    val thumbnailed = thumbnail(cropSource)(width)
+    val corrected   = correctColour(thumbnailed)(iccColourSpace, colourModel)
+    val converted   = applyOutputProfile(corrected, optimised = true)
+    val stripped    = stripMeta(converted)
+    val profiled    = applyOutputProfile(stripped, optimised = true)
+    val unsharpened = unsharp(profiled)(thumbUnsharpRadius, thumbUnsharpSigma, thumbUnsharpAmount)
+    val qualified   = quality(unsharpened)(qual)
+    val addOutput   = {file:File => addDestImage(qualified)(file)}
     for {
-      outputFile <- createTempFile(s"thumb-", ".jpg", tempDir)
-      cropSource  = addImage(sourceFile)
-      thumbnailed = thumbnail(cropSource)(width)
-      corrected   = correctColour(thumbnailed)(iccColourSpace, colourModel)
-      converted   = applyOutputProfile(corrected, optimised = true)
-      stripped    = stripMeta(converted)
-      profiled    = applyOutputProfile(stripped, optimised = true)
-      unsharpened = unsharp(profiled)(thumbUnsharpRadius, thumbUnsharpSigma, thumbUnsharpAmount)
-      qualified   = quality(unsharpened)(qual)
-      addOutput   = addDestImage(qualified)(outputFile)
-      _          <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains(Tiff))
-    } yield outputFile
+      outputFile <- createTempFile(s"thumb-", thumbMimeType.fileExtension, tempDir)
+      _          <- runConvertCmd(addOutput(outputFile), useImageMagick = sourceMimeType.contains(Tiff))
+    } yield (outputFile, thumbMimeType)
   }
 
-  def transformImage(sourceFile: File, sourceMimeType: Option[MimeType], tempDir: File): Future[File] = {
+  /**
+    * Given a source file containing a file which requires optimising to make it suitable for viewing in
+    * a browser, construct a new image file in the provided temp directory, and return
+    * * the file with metadata about it.
+    * @param sourceFile File containing browser viewable (ie not too big or colourful) image
+    * @param sourceMimeType Mime time of browser viewable file
+    * @param tempDir Location to create optimised file
+    * @return The file created and the mimetype of the content of that file, in a future.
+    */
+  def transformImage(sourceFile: File, sourceMimeType: Option[MimeType], tempDir: File): Future[(File, MimeType)] = {
     for {
-      outputFile      <- createTempFile(s"transformed-", ".png", tempDir)
+      // png suffix is used by imagemagick to infer the required type
+      outputFile      <- createTempFile(s"transformed-", Png.fileExtension, tempDir)
       transformSource = addImage(sourceFile)
       addOutput       = addDestImage(transformSource)(outputFile)
       _               <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains(Tiff))
-      _               <- checkForOutputFileChange(outputFile)
-    } yield outputFile
+      extension       <- checkForOutputFileChange(outputFile)
+    } yield (outputFile, MimeType(extension))
```

Comment:
I'm still confused by this? Is the valid input file for `checkForOutputFileChange` somehting like `<filename>.<validMimeType>`? Granted, [MimeType](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/model/MimeType.scala#L28-L35) seems to deal with a subset of extensions and will throw an exception otherwise - but that seems to happen not becaue it is designed to deeal with file extensions but rather because of a historical data anomaly.

Reply:
It's not really being used as a mimetype, it's just changing behaviour if it's a tiff.

Any input file supported by imagemagick is fine.